neo4j-store/src/main/kotlin/org/ostelco/prime/storage/graph/Neo4jStore.kt

```diff
@@ -806,26 +806,43 @@ object Neo4jStoreSingleton : GraphStore {
         }
 
         return IO {
-            Either.monad<StoreError>()
-                    .binding {
-                        simProfiles.bind().map { simProfile ->
-                            val regionId = (regionCode ?: map[simProfile.id])
-                                    ?: ValidationError(type = simProfileEntity.name, id = simProfile.iccId, message = "SimProfile not linked to any region")
-                                            .left()
-                                            .bind()
-
-                            val simEntry = simManager.getSimProfile(
-                                    hlr = hssNameLookup.getHssName(regionId),
-                                    iccId = simProfile.iccId)
-                                    .mapLeft { NotFoundError(type = simProfileEntity.name, id = simProfile.iccId) }
-                                    .bind()
-                            org.ostelco.prime.model.SimProfile(
-                                    iccId = simProfile.iccId,
-                                    alias = simProfile.alias,
-                                    eSimActivationCode = simEntry.eSimActivationCode,
-                                    status = simEntry.status)
-                        }
-                    }.fix()
+            Either.monad<StoreError>().binding {
+                simProfiles.bind().map { simProfile ->
```

Comment:
Which map is this? Is it the arrow one, or the kotlin one,?

Reply:
This is `map` on Collection.
After `.bind()`, you get the "right" value of Either, which is Collection<SimProfile>.